### PR TITLE
HDDS-6457. Add freon command to run r/w mix workload using FS

### PR DESCRIPTION
**Pull request by tanvipenumudy**
## What changes were proposed in this pull request?

To develop a new Ozone Freon tool command where one can configure or pass an argument to generate/run a synthetic read and write operation mix workload on a bucket and print the lock read/write waiting/held time metrics.

**Step 1:**  Create `fileCountForRead` (defaultValue = 1000)  files under _rootPath/readPath_ directory
**Step 2:**  Start `readThreadCount` (defaultValue = 90) concurrent read threads
	&emsp;each new Thread (() -> {
		&emsp;&emsp;for(1...`numOfReadOperations` (defaultValue = 50))
			&emsp;&emsp;&emsp;fileSystem.listStatus(_rootPath/readPath_)
	&emsp;}
**Step 3:**  Start `writeThreadCount` (defaultValue = 10) concurrent write threads
	&emsp;each new Thread (() -> {
		&emsp;&emsp;for(1...`numOfWriteOperations` (defaultValue = 10))
			&emsp;&emsp;&nbsp;createFiles(_rootPath/writePath/file-random-keypath_)
	&emsp;}

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6457

## How was this patch tested?

Added integration tests.

### CODE REVIEW

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OmBucketReadWriteOps.java [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    mixinStandardHelpOptions = true,
+    showDefaultValues = true)
+
+public class OmBucketReadWriteOps extends BaseFreonGenerator
+    implements Callable<Void> {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(OmBucketReadWriteOps.class);
+
+  @Option(names = {"-P", "root-path", "--rootPath"},
```
Comment (JyotinderSingh) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OmBucketReadWriteOps.java:57]:
```suggestion
  @Option(names = {"-P", "--root-path", "--rootPath"},
```
minor typo

Comment (tanvipenumudy) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OmBucketReadWriteOps.java]: Fixed the typo, thank you!

**Comment by tanvipenumudy** [comment]
Hi @siddhantsangwan, @rakeshadr it would be great if you can review the patch, thanks!

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OmBucketReadWriteOps.java [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+ */
+@Command(name = "obrwo",
+    aliases = "om-bucket-read-write-ops",
+    description = "Creates files, performs respective read/write " +
+        "operations to measure lock performance.",
+    versionProvider = HddsVersionProvider.class,
+    mixinStandardHelpOptions = true,
+    showDefaultValues = true)
+
+public class OmBucketReadWriteOps extends BaseFreonGenerator
```
Comment (rakeshadr) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OmBucketReadWriteOps.java:56]: Please modify the freon to use ObjectStore APIs as this is used to evaluate performance of `OBS` bucket layout keys. Please refer OzoneClientKeyGenerator or OzoneClientKeyRemover, you can use bucket#listKeys and bucket#createKey.

Also, you can rename the freon tool to `OmBucketReadWriteKeyOps`

Thanks @mukul1987 for the offline discussion.

Comment (rakeshadr) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OmBucketReadWriteOps.java:56]: On a second thought, you can even extend your implementation to run o3fs or ObjectStore API operations based on the given bucket's layout.

```
If (bucketLayout.isFileSystemOptimized()) {
    // o3fs APIs
else {
   // ObjectStore APIs
}
```

Comment (rakeshadr) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OmBucketReadWriteOps.java:56]: Another idea can be like, two different tools `obrwk` and `obrwf`, better not to complicate things.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OmBucketReadWriteFileOps.java [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+  @Option(names = {"-l", "--name-len", "--nameLen"},
+      description =
+          "Length of the random name of directory you want to create. Full " +
+              "name --nameLen will be removed in later versions.",
+      defaultValue = "10")
+  private int length;
+
+  @Option(names = {"-c", "--total-thread-count", "--totalThreadCount"},
+      description = "Total number of threads to be executed. Full name " +
+          "--totalThreadCount will be removed in later versions.",
```
[the remark this answers is not in the document]
Comment (tanvipenumudy) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OmBucketReadWriteFileOps.java]: I have omitted the camelCase option names, thank you for the comment!

**Comment by rakeshadr** [comment]
Thanks @adoroszlai  @JyotinderSingh for the reviews!

**Pull request merged**